FE-958 - Hibana style cleanup for Signals health score, spam traps, and engagement pages

nicklemmon opened the pull request:
[FE-958](https://jira.int.messagesystems.com/browse/FE-958)

### What Changed

- Fixed `Callout` component bug by conditionally rendering the `title` content 
- Updated React Testing Library ESLint plugin to give more accurate warnings, re: using `queryBy*` queries
- Updated a lot of `/signals` components to look appropriate for Hibana
- Fixed a bug with the Hibana version of `SortLabel` by leveraging the `align` prop to align things via flexbox

### How To Test
- Verify the following routes look nice and behave themselves for both the OG and Hibana themes:
  - /signals/health-score
  - /signals/spam-traps
  - /signals/engagement
  - /signals/health-score/:facet/:facetId
  - /signals/spam-traps/:facet/:facetId
  - /signals/engagement/cohorts/:facet/:facetId
  - /signals/engagement/engagement-rate/:facet/:facetId
  - /signals/engagement/unsubscribes/:facet/:facetId
  - /signals/engagement/complaints/:facet/:facetId

### To Do
- [x] Fix all the tests 😢 
- [x] Incorporate feedback

nicklemmon commented on src/components/summaryTable/Head.js:
```diff
@@ -37,7 +37,7 @@ function HibanaHead({ columns, onSort, order }) {
       <Table.Row header>
         {columns.map(({ align = 'left', dataKey, label, sortable, width }) => (
-          <Table.HeaderCell style={{ 'text-align': align }} key={`header-${dataKey}`} width={width}>
+          <Table.HeaderCell style={{ textAlign: align }} key={`header-${dataKey}`} width={width}>
```
The prior implementation was throwing some errors in the console - inline styles are kind of weird...

bkemper commented on src/components/charts/TooltipMetric.js:
```diff
@@ -17,12 +18,12 @@ export const OGTooltipMetric = ({ color = '#6e6e73', description, label, value }
   label = '',
   value = '',
 }) => (
-  <Box position="relative" display="inline-block" width={1} height="200">
```
`width={1}` 😕

bkemper commented on cypress/.eslintrc:
```diff
@@ -5,6 +5,8 @@
     // sometimes only another browser interaction occurs rather than an assertion.
     // Docs for the rule: https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/valid-expect-in-promise.md
-    "jest/valid-expect-in-promise": "off"
+    "jest/valid-expect-in-promise": "off",
```
Maybe we should move jest eslint configuration to src/.eslintrc?

nicklemmon replied:
*Should* be easy enough - let me take a stab at it - and if it's too time consuming I'll bail

bkemper replied:
Bail!

bkemper commented on src/components/collection/SortLabel.js:
```diff
@@ -3,7 +3,7 @@ import classNames from 'classnames';
 import { ArrowDropDown, ArrowDropUp } from '@sparkpost/matchbox-icons';
 import { Box, Text, UnstyledLink } from 'src/components/matchbox';
 import OGStyles from './SortLabel.module.scss';
-import HibanaStyles from './SortLabelHibana.module.scss';
```
:) I have been doing it the other way.

nicklemmon replied:
🤷 Hard to do it consistently when there is an acronym in the mix!

bkemper commented on src/pages/signals/UnsubscribeRateByCohortPage.js:
```diff
@@ -24,7 +23,6 @@ import _ from 'lodash';
 import SpamTrapsPreview from './components/previews/SpamTrapsPreview';
 import HealthScorePreview from './components/previews/HealthScorePreview';
```
Shouldn't we remove these to match health score and spam traps?

bkemper commented on src/components/charts/TooltipMetric.js:
```diff
@@ -17,12 +18,12 @@ export const OGTooltipMetric = ({ color = '#6e6e73', description, label, value }
+  <Box position="relative" display="inline-block" height="500">
     <Box display="flex" justifyContent="space-between">
       <Box bg={color} marginRight="100" borderRadius={8} size={16} marginTop="100" />
       <Box display="flex" height="100" flexDirection="column" flex="1 0 0">
```
I think this is the right tooltip, but we are missing some spacing between label and value.

<img width="414" alt="Screen Shot 2020-05-13 at 11 09 21 AM" src="https://user-images.githubusercontent.com/1335605/81832951-1b19a380-950d-11ea-99c7-aaddc00c50f6.png">

bkemper replied:
This looks good now!

nicklemmon commented:
@bkemper Ready for review - all tests _should_ be passing now. Cypress did catch a bug I had to address. We need that bad boy everywhere.

bkemper commented on src/components/charts/TooltipMetric.js:
```diff
@@ -17,12 +18,12 @@ export const OGTooltipMetric = ({ color = '#6e6e73', description, label, value }
   value = '',
 }) => (
-  <Box position="relative" display="inline-block" width={1} height="200">
+  <Box position="relative" display="inline-block" height="500">
```
I think engagement recency is the only chart with tooltip descriptions. 

<img width="618" alt="Screen Shot 2020-05-13 at 11 48 13 AM" src="https://user-images.githubusercontent.com/1335605/81835046-c6c3f300-950f-11ea-8f42-dcfcee351a09.png">

bkemper replied:
Looks good now.

bkemper commented on src/pages/signals/SpamTrapPage.js:
```diff
@@ -15,30 +16,24 @@ import withDateSelection from './containers/withDateSelection';
+export function SpamTrapPage(props) {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  const [calculation, setCalculation] = useState('relative');
```
👏

bkemper commented on src/pages/signals/SpamTrapPage.js:
```diff
@@ -168,48 +169,49 @@ export class SpamTrapPage extends Component {
-          </div>
+          {!loading && (
+            <Box as={Panel} sectioned>
+              <div style={{ height: 'fill-parent' }}>
```
Is "fill-parent" a valid height value?

nicklemmon replied:
Nope...not sure where that came from 🤷

bkemper commented on src/pages/signals/components/ActionsHibana.module.scss:
```diff
@@ -0,0 +1,49 @@
+  p {
+    font-size: font-size(300);
+    color: color(gray, 900);
+    line-height: line-height(300);
```
This shouldn't be needed.  It matches the default for <p>.

<img width="1390" alt="Screen Shot 2020-05-13 at 12 07 35 PM" src="https://user-images.githubusercontent.com/1335605/81837106-5ff40900-9512-11ea-93a7-dc8ebba4d480.png">

bkemper commented on src/pages/signals/components/charts/divergingBar/DivergingBar.js:
```diff
@@ -28,46 +43,47 @@ class DivergingBar extends Component {
           x2={width}
           width={width}
-          fill='#5DCFF5'
+          fill="#5DCFF5"
```
:( 1b62e062a

nicklemmon replied:
yeeeah it introduced a bug Cypress caught. I'm still not entirely sure why, but rather just leave it than do more digging at this point.

bkemper commented on src/pages/signals/components/dataCells/tests/__snapshots__/WoWHeaderCell.test.js.snap:
```diff
@@ -4,6 +4,7 @@ exports[`WoWHeaderCell renders 1`] = `
 <Tooltip
   content="Week Over Week Change"
   dark={true}
+  id="wow-header-cell-1"
```
I don't think it will be a problem for this test, but for future tests.  I setup a default mock for `useUniqueId` to always return `-0`?

```
jest.mock('src/hooks/useUniqueId/useUniqueId');
```

bkemper commented on src/pages/signals/components/filters/FacetFilter.js:
```diff
@@ -56,25 +56,25 @@ export class FacetFilter extends React.Component {
 
     return (
-      <Grid.Column lg={facet ? 4 : 3} xl={facet ? 5 : 3}>
+      <Grid.Column lg={facet ? 7 : 3} xl={facet ? 7 : 3}>
```
The xl prop should be needed since it sets the same values as lg.

bkemper commented on src/components/charts/TooltipMetric.js:
```diff
@@ -16,33 +17,50 @@ export const OGTooltipMetric = ({ color = '#6e6e73', description, label, value }
+  <Box display="flex" alignItems="flex-start" justifyContent="flex-start" width="100%">
+    <Box
+      backgroundColor={color}
+      border="1px solid white" // todo, yuck
```
haha... You are willing to take ownership of this comment.  😄

nicklemmon commented:
@bkemper Ready for another pass. Addressed your feedback, got colors working, and got tests working again (that last part was the largest amount of effort, unfortunately...)
